src/main/java/de/tum/cit/aet/artemis/core/service/course/CourseAccessService.java

```diff
@@ -0,0 +1,232 @@
+    }
+
+    /**
+     * Add multiple users to the course so that they can access it
+     * The passed list of UserDTOs must include at least one unique user identifier (i.e. registration number OR email OR login)
+     * <p>
+     * This method first tries to find the user in the internal Artemis user database (because the user is probably already using Artemis).
+     * In case the user cannot be found, it additionally searches the connected LDAP in case it is configured.
+     *
+     * @param courseId    the id of the course
+     * @param studentDTOs the list of students (with at least registration number)
+     * @param courseGroup the group the students should be added to
+     * @return the list of students who could not be enrolled in the course, because they could NOT be found in the Artemis database and could NOT be found in the TUM LDAP
+     */
+    public List<StudentDTO> registerUsersForCourseGroup(Long courseId, List<StudentDTO> studentDTOs, String courseGroup) {
+        var course = courseRepository.findByIdElseThrow(courseId);
+        if (course.getLearningPathsEnabled()) {
+            course = courseRepository.findWithEagerCompetenciesAndPrerequisitesByIdElseThrow(course.getId());
+        }
+        String courseGroupName = course.defineCourseGroupName(courseGroup);
+        Role courseGroupRole = Role.fromString(courseGroup);
+        List<StudentDTO> notFoundStudentsDTOs = new ArrayList<>();
+        for (var studentDto : studentDTOs) {
+            var optionalStudent = userService.findUserAndAddToCourse(studentDto.registrationNumber(), studentDto.login(), studentDto.email(), courseGroupName);
+            if (optionalStudent.isEmpty()) {
+                notFoundStudentsDTOs.add(studentDto);
+            }
+            else if (courseGroupRole == Role.STUDENT && course.getLearningPathsEnabled()) {
+                final Course finalCourse = course;
+                learnerProfileApi.ifPresent(api -> api.createCourseLearnerProfile(finalCourse, optionalStudent.get()));
+                learningPathApi.ifPresent(api -> api.generateLearningPathForUser(finalCourse, optionalStudent.get()));
+            }
+        }
+
+        return notFoundStudentsDTOs;
+    }
+
+    /**
+     * Returns all users in a course that belong to the given group
+     *
+     * @param course    the course
+     * @param groupName the name of the group
+     * @return list of users
+     */
+    @NotNull
+    public ResponseEntity<Set<User>> getAllUsersInGroup(Course course, String groupName) {
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+        var usersInGroup = userRepository.findAllByDeletedIsFalseAndGroupsContains(groupName);
+        usersInGroup.forEach(user -> {
+            // explicitly set the registration number
+            user.setVisibleRegistrationNumber(user.getRegistrationNumber());
+        });
```

Comment:
I do see that the registration number is set explicitly, but why is that the case? I do not think that the current comment adds value and should rather explain why we need /want to set it explicitly